sandbox/anvi-script-filter-hmm-hit-table

```diff
@@ -0,0 +1,223 @@
+
+        if target[0] != 'AA':
+            raise ConfigError("The hmm-source %s is not for amino acid sequences. "
+                              "anvi-script-filter-hmm-hit-table currently can only work with hmm-sources from protein sequences."  % (self.hmm_source))
+
+
+    def import_domtblout(self):
+
+        # hmmsearch results so the queries are the hmm models and the targets are the ORFs
+
+        col_info = [
+            ('gene_callers_id',  int),   # target name
+            ('target_accession', str),   # target accession
+            ('gene_length',      int),   # tlen
+            ('hmm_name',         str),   # query name
+            ('hmm_id',           str),   # accession
+            ('hmm_length',       int),   # qlen
+            ('evalue',           float), # E-value (full sequence)
+            ('bitscore',         float), # score (full sequence)
+            ('bias',             float), # bias (full sequence)
+            ('match_num',        int),   # # (this domain)
+            ('num_matches',      int),   # of (this domain)
+            ('dom_c_evalue',     float), # c-Evalue (this domain)
+            ('dom_i_evalue',     float), # i-Evalue (this domain)
+            ('dom_bitscore',     str),   # score (this domain)
+            ('dom_bias',         float), # bias (this domain)
+            ('hmm_start',        int),   # from (hmm coord)
+            ('hmm_stop',         int),   # to (hmm coord)
+            ('gene_start',       int),   # from (ali coord)
+            ('gene_stop',        int),   # to (ali coord)
+            ('env_to',           str),   # from (env coord)
+            ('env_from',         str),   # to (env coord)
+            ('mean_post_prob',   float), # acc
+        ]
+
+        try:
+            colnames_coltypes_list = list(zip(*col_info))
+            colnames_coltypes_dict = dict(zip(colnames_coltypes_list[0], colnames_coltypes_list[1]))
+
+            self.df=pd.read_csv(self.domtblout,
+                            delim_whitespace=True,
+                            comment='#',
+                            names=colnames_coltypes_list[0],
+                            dtype=colnames_coltypes_dict,
+                            header=None,
+                            index_col=False)
+        except Exception as e:
+            print(e)
+            raise ConfigError("Doesn't look like a --domtblout")
+
+        self.df['query_coverage'] = ((self.df['hmm_stop'] - self.df['hmm_start'])/ self.df['hmm_length'])
+        self.df['target_coverage'] = ((self.df['gene_stop'] - self.df['gene_start'])/ self.df['gene_length'])
+
+
+    def filter_domtblout(self):
+
+        # Filtering conditions
+        if self.target_coverage:
+            df_filtered = self.df[self.df['target_coverage'] > float(self.target_coverage)]
+
+        if self.query_coverage:
+            df_filtered = self.df[self.df['query_coverage'] > float(self.query_coverage)]
```

Comment:
If the user provides both target coverage and query coverage filtering values, these two conditionals will run in addition to the one below, which _I think_ makes the below if statement redundant? I didn't think about it too much so I might be wrong, but I just wanted to bring it up so you can take a look and make sure that is what you want to happen :)

Reply:
I fixed this